### ISSUE TITLE
Support new board manifest `_pins.yaml`

### ISSUE THREAD

**Issue by juliasilge** [rstudio/pins-python#174]
In rstudio/pins-r#661 we [added support for a manifest file](https://pins.rstudio.com/dev/reference/write_board_manifest.html). In pins for Python, we now need to:

- not identify the new `_pins.yaml` file at the root of the board as a pin
- allow folks to write a manifest file from Python

The first one is high priority as we plan to not release pins for R until this gets implemented. The second is low priority and we can wait on it until whatever is a good time.

**Comment by juliasilge** [comment]
@juliasilge it appears adding a `pins.txt` file as a manifest breaks `pin_search` on python pins. The reason is that in the python version `pin_list()` returns the filename, and `pin_search` attempts to retrieve metadata for every pin as part of search.

Here's the relevant R pins [pins_search code fetching metadata](https://github.com/rstudio/pins-r/blob/b0b6a47f654a1b55159991aeb167dbfcbe6ea8a5/R/pin_search.R#L32-L34) (which works fine with the manifest file).

wdyt is a good move here? I can modify pins-python to ignore `pins.txt` files, but I'm wondering what the right way sequence is for releasing..

### edit: how I tested

Using an .env file with credentials filled in from [.env.dev](https://github.com/rstudio/pins-python/blob/main/.env.dev).

In python:

```python
from dotenv import load_dotenv
load_dotenv()
from pins import board_s3
from pins.tests.helpers import BoardBuilder

bb = BoardBuilder("s3")
board = bb.create_tmp_board("pins/tests/pins-compat")

# list path to temporary pins bucket, for copying into R
board.board
```

Switching to R:

```R
board <- board_s3(<bucket_from_above>, prefix="<prefix_from_above>/")

# note that this also prints a warning
board %>% board_manifest()
```

Back in python:

```python
board.pin_search("abc")
```

_Originally posted by @machow in https://github.com/rstudio/pins-r/issues/661#issuecomment-1307576924_

**Comment by machow** [comment]
v0.7.3 shouldn't identify `_pins.yaml` as a pin. Still need to add an analogue to R pin's `write_board_manifest()`
